nixos/modules/services/security/bitwarden-directory-connector.nix

```
@@ -0,0 +1,328 @@
+            default = "cn";
+          };
+          groupFilter = mkOption {
+            type = types.str;
+            description = lib.mdDoc "Filter for groups.";
+            example = "(cn=sales)";
+            default = "";
+          };
+        };
+      };
+    };
+
+    env = {
+      ldap = mkOption rec {
+        type = types.str;
+        description = "Auth for the LDAP, has value defined in {option}`pw_env";
+      };
+      bitwarden = mkOption rec {
+        type = types.str;
+        description = "Auth for Bitwarden, has BW_CLIENTID and BW_CLIENTSECRET";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    users.groups."${cfg.user}" = {};
+
+    users.users."${cfg.user}" = {
+      createHome = true;
+      isSystemUser = true;
+      home = "${cfg.directory}";
+      group = "${cfg.user}";
+      homeMode = "711";
+    };
+
+    systemd = {
+      timers.bitwarden_directory_connector = {
+        description = "Sync timer for Bitwarden Directory Connector";
+        wantedBy = ["timers.target"];
+        partOf = ["bitwarden_directory_connector.service"];
+        timerConfig = {
+          OnCalendar = cfg.interval;
+          Unit = "bitwarden_directory_connector.service";
+          Persistent = true;
+        };
+      };
+
+      services.bitwarden_directory_connector = {
+        description = "Main process for Bitwarden Directory Connector";
+        wantedBy = ["multi-user.target"];
+        after = ["network-online.target"];
+        wants = [];
+        path = [pkgs.jq];
+
+        environment = {
+          BITWARDENCLI_CONNECTOR_APPDATA_DIR = cfg.directory;
+          BITWARDENCLI_CONNECTOR_PLAINTEXT_SECRETS = "true";
+        };
+
+        serviceConfig = {
+          Type = "oneshot";
+          User = "${cfg.user}";
+          Group = "${cfg.user}";
+          ExecStartPre = pkgs.writeShellScript "bitwarden_directory_connector-config" ''
+            # create the config file
+            ${cfg.package}/bin/${cfg.binary_name} data-file
+
+            ${cfg.package}/bin/${cfg.binary_name} config server ${cfg.domain}
+
+            # now login to set credentials
+            ${cfg.package}/bin/${cfg.binary_name} login
+
+            jq '.authenticatedAccounts[0] as $account
+              | .[$account].directoryConfigurations.ldap |= $ldap_data
+              | .[$account].directorySettings.organizationId |= $orgID
+              | .[$account].directorySettings.sync |= $sync_data' \
+              --argjson ldap_data ${escapeShellArg ldap_data} \
+              --arg orgID "''${BW_CLIENTID//organization.}" \
+              --argjson sync_data ${escapeShellArg sync_data} \
+              ${escapeShellArg cfg.directory}/data.json \
+              > ${escapeShellArg cfg.directory}/data.json.tmp
+
+            mv -f -- ${escapeShellArg cfg.directory}/data.json.tmp ${escapeShellArg cfg.directory}/data.json
+            chmod 600 ${escapeShellArg cfg.directory}/data.json
```

There's still a small window of readability here. How about this, just to be 100% safe:
```suggestion
            touch ${escapeShellArg cfg.directory}/data.json.tmp
            chmod 600 ${escapeShellArg cfg.directory}/data.json.tmp

            jq '.authenticatedAccounts[0] as $account
              | .[$account].directoryConfigurations.ldap |= $ldap_data
              | .[$account].directorySettings.organizationId |= $orgID
              | .[$account].directorySettings.sync |= $sync_data' \
              --argjson ldap_data ${escapeShellArg ldap_data} \
              --arg orgID "''${BW_CLIENTID//organization.}" \
              --argjson sync_data ${escapeShellArg sync_data} \
              ${escapeShellArg cfg.directory}/data.json \
              > ${escapeShellArg cfg.directory}/data.json.tmp

            mv -f -- ${escapeShellArg cfg.directory}/data.json.tmp ${escapeShellArg cfg.directory}/data.json
```

---

Moved it up to where the config file is created, fits better there.